src/trello/trello_api.py

```diff
@@ -0,0 +1,346 @@
+def get_trello_list_id(list_name: str, include_archived: bool = False) -> Optional[str]:
+    """
+    Fetch a Trello list ID by name.
+
+    Args:
+        list_name (str): The name of the Trello list to fetch.
+        include_archived (bool): Whether to include archived lists in the search. Defaults to False.
+
+    Returns:
+        Optional[str]: The Trello list ID if found, else None.
+
+    Raises:
+        Exception: If the lists cannot be fetched due to an API error.
+    """
+    # Fetch all lists (both open and closed if include_archived is True)
+    filter_value = 'all' if include_archived else 'open'
+    url_all_lists = f"https://api.trello.com/1/boards/{TRELLO_BOARD_ID}/lists"
+    query_params = {
+        'key': TRELLO_API_KEY,
+        'token': TRELLO_TOKEN,
+        'filter': filter_value
+    }
+    response_all = requests.get(url_all_lists, params=query_params)
+
+    if response_all.status_code == 200:
+        lists = response_all.json()
+        for lst in lists:
+            if lst['name'].lower() == list_name.lower():
+                return lst['id']
+        # List not found
+        return None
+    else:
+        raise Exception(f"Failed to fetch lists from Trello: {response_all.status_code}, {response_all.text}")
+# Function to create a new Trello list
+def create_trello_list(list_name: str) -> str:
+    """
+    Create a new Trello list.
+
+    Args:
+        list_name (str): The name of the Trello list to create.
+
+    Returns:
+        str: The Trello list ID.
+
+    Raises:
+        Exception: If the list cannot be created due to an API error.
+    """
+    create_list_url = f"https://api.trello.com/1/boards/{TRELLO_BOARD_ID}/lists"
+    query = {
+        'key': TRELLO_API_KEY,
+        'token': TRELLO_TOKEN,
+        'name': list_name,
+        'pos': 'bottom'
+    }
+    create_response = requests.post(create_list_url, params=query)
+    if create_response.status_code == 200:
+        return create_response.json()['id']
+    else:
+        raise Exception(f"Failed to create list '{list_name}': {create_response.status_code}, {create_response.text}")
+
+# Add a card to a specified list in Trello with an optional due date
+def add_card_to_trello(card_name: str, list_name: str, due_date: str = None) -> dict:
+    """
+    Add a card to a specified Trello list with an optional due date.
+
+    Args:
+        card_name (str): The name of the card to add.
+        list_name (str): The name of the Trello list to add the card to.
+        due_date (str, optional): The due date for the card in ISO 8601 format. Defaults to None.
+
+    Returns:
+        dict: The created card's details.
+
+    Raises:
+        Exception: If the card cannot be added due to an API error.
+    """
+    list_id = get_trello_list_id(list_name)
+    if list_id is None:
+        # List does not exist, so create it
+        list_id = create_trello_list(list_name)
+
+    url = f"https://api.trello.com/1/cards"
```

Comment:
Use more specific names for url in each function, such as create_card_url

Reply:
This has been fixed in the new version.